service/controller/resource/statuscondition/create_test.go

```diff
@@ -715,15 +703,57 @@ func TestComputeVersionChange(t *testing.T) {
 					unittest.GetCreatingCondition(90),
 				},
 				Versions: []infrastructurev1alpha2.CommonClusterStatusVersion{
+					unittest.GetVersion(10, "8.7.6"),
 					unittest.GetVersion(60, "8.7.5"),
-					unittest.GetVersion(60, "8.7.6"),
 				},
 			},
 			nodesReady:     true,
 			desiredVersion: "8.7.6",
 
 			expectedResult: false,
 		},
+		// the cluster has not transitioned yet
```

Comment:
This works now, too! Previously, the version was already set before the update was done.